### PR TITLE
[mod] upgrade to Caddy v2

### LINKED ISSUE #35: Any plans to support caddy v2?
Hello! As you can know, Caddy v2 is out. Do you plan to migrate to it? Thanks in beforehand

### LINKED ISSUE #37: Changing port is not reflected in parts of the Searx interface
I changed SEARX_HOSTNAME in .env to localhost:8888 but this isn't reflected in some parts of the searx interface. For example the search link is still presented as this (note it is not `localhost:8888`): `https://localhost/?q=asdf&categories=general&language=en-US`
Morty links are also broken in this way.

### PR DESCRIPTION
**Pull request by dalf**
* Use docker image caddy/caddy:2-alpine
* Caddyfile: remove "limits 10KB"
* Caddyfile: URL /filtron/rules removes (filtron API still availabled on http://localhost:4041/rules )
* caddy storage are docker volumes (caddy-data and caddy-config). ```start.sh``` and ```stop.sh``` have been modified to keep these volumes.
* .env: Remove SEARX_PROTOCOL, SEARX_TLS, FILTRON_USER and FILTRON_PASSWORD variables.
* docker-compose.yml: filtron and morty listen on 127.0.0.1 (related to #38)

* Fix #37: settings ```SEARX_HOSTNAME=localhost:8888``` works as expected (https connection)

### CODE REVIEW

**Comment by dalf** [comment]
Test instance: https://a.searx.space

Everything seems to work as expected.
@unixfox , I don't know if you have time to have a quick look at this PR, at least to check if it does not stray from standards more than before.

File: docker-compose.yaml [hunk 1]
```diff
@@ -4,23 +4,17 @@ services:
...
-      - 443:443
+    image: caddy/caddy:2-alpine
     network_mode: host
-    command: -email ${LETSENCRYPT_EMAIL} -agree=${LETSENCRYPT_AGREE} -log stdout -host ${SEARX_HOSTNAME} -conf /etc/Caddyfile
+    command: caddy run -config /etc/caddy/Caddyfile
     volumes:
-      - ./Caddyfile:/etc/Caddyfile:rw
-      - ./caddy:/root/.caddy:rw
-      - ./srv:/srv:rw
+      - ./Caddyfile:/etc/caddy/Caddyfile:rw
```
Comment (unixfox) [inline docker-compose.yaml]: could be set to `./Caddyfile:/etc/caddy/Caddyfile:ro` because the config doesn't change

File: docker-compose.yaml [hunk 2]
```diff
@@ -4,23 +4,17 @@ services:
 
   caddy:
     container_name: caddy
-    image: abiosoft/caddy:1.0.3-no-stats
-    ports:
-      - 80:80
-      - 443:443
+    image: caddy/caddy:2-alpine
```
Comment (unixfox) [inline docker-compose.yaml]: wrong docker repo. it's `caddy:2-alpine`

File: docker-compose.yaml [hunk 3]
```diff
@@ -4,23 +4,17 @@ services:
...
     network_mode: host
-    command: -email ${LETSENCRYPT_EMAIL} -agree=${LETSENCRYPT_AGREE} -log stdout -host ${SEARX_HOSTNAME} -conf /etc/Caddyfile
+    command: caddy run -config /etc/caddy/Caddyfile
```
Comment (unixfox) [inline docker-compose.yaml]: why overriding the default command? the default Dockerfile already include the correct command: https://github.com/caddyserver/caddy-docker/blob/master/2.1/alpine/Dockerfile#L65

File: docker-compose.yaml (hunk 1 above)
Comment (dalf) [reply docker-compose.yaml]: Updated.

File: docker-compose.yaml (hunk 2 above)
Comment (dalf) [reply docker-compose.yaml]: Thank you!

File: docker-compose.yaml (hunk 3 above)
Comment (dalf) [reply docker-compose.yaml]: removed.
(the other docker image seems to require this).

**Comment by dalf** [comment]
Thank you @unixfox

**Pull request merged**